File: exampleProjects/java-benchmarks/src/jmh/java/io/morethan/javabenchmarks/showcase/QuickBenchmark.java
package io.morethan.javabenchmarks.showcase;

import java.util.concurrent.TimeUnit;

import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.BenchmarkMode;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Measurement;
import org.openjdk.jmh.annotations.Mode;
import org.openjdk.jmh.annotations.OutputTimeUnit;
import org.openjdk.jmh.annotations.Warmup;

/**
 * Benchmark which executes very fast to test the pipeline.
 */
@Fork(value = 1)
@Warmup(iterations = 0)
@Measurement(iterations = 1)
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
public class QuickBenchmark {

    @Benchmark
    public void sleep50Milliseconds() throws InterruptedException {
        Thread.sleep(50);
    }

    @Benchmark
    public void sleep100Milliseconds() throws InterruptedException {
        Thread.sleep(100);
    }

}

File: exampleProjects/kotlin-benchmarks/src/jmh/kotlin/io/morethan/kotlinbenchmarks/QuickBenchmark.kt
package io.morethan.kotlinbenchmarks

import org.openjdk.jmh.annotations.State
import org.openjdk.jmh.annotations.Scope
import org.openjdk.jmh.annotations.BenchmarkMode
import org.openjdk.jmh.annotations.Mode
import org.openjdk.jmh.annotations.Benchmark
import org.openjdk.jmh.annotations.Fork
import org.openjdk.jmh.annotations.Measurement
import org.openjdk.jmh.annotations.Warmup
import org.openjdk.jmh.annotations.OutputTimeUnit
import java.util.concurrent.TimeUnit

@Fork(value = 1)
@Warmup(iterations = 0)
@Measurement(iterations = 1)
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
open public class QuickBenchmark {
    
    
    @Benchmark
    public fun sleep50Milliseconds() {
        Thread.sleep(50);
    }

    @Benchmark
    public fun sleep100Milliseconds() {
        Thread.sleep(100);
    }

}